src/sinol_make/commands/run/__init__.py

```diff
@@ -631,18 +635,16 @@ def run_solution(self, data_for_execution: ExecutionData):
         Run an execution and return the result as ExecutionResult object.
         """
 
-        (name, executable, test, time_limit, memory_limit, timetool_path, execution_dir) = data_for_execution
+        (name, executable, test, output_test, time_limit, memory_limit, timetool_path, execution_dir) = data_for_execution
         file_no_ext = paths.get_executions_path(name, package_util.extract_test_id(test, self.ID))
         output_file = file_no_ext + ".out"
         result_file = file_no_ext + ".res"
-        hard_time_limit_in_s = math.ceil(2 * time_limit / 1000.0)
+        hard_time_limit = math.ceil(2 * time_limit / 1000.0)
```

Comment:
Is this 2 times time limit in any way tested to be enough? I have an example, where sio2jail runs 1.5 times longer than time measured by the jail. If not, given "worse" program or slower machine, this hard time limit may be exceeded, without exceeding time_limit.

Reply:
It seems to work well in real sioworkers. But I guess on some systems it may fail

Reply:
This may be another discrepancy between upstream sioworkers and those on sio2.mimuw, but the first repo contains a multiplier of ***16*** with an additional second, see [here](https://github.com/sio2project/sioworkers/blob/9137c6a3726dd6de422dc79ed16319d782ec78f4/sio/workers/executors.py#L598-L599) and [here](https://github.com/sio2project/sioworkers/blob/9137c6a3726dd6de422dc79ed16319d782ec78f4/sio/workers/executors.py#L619-L627).

Reply:
Okay, I didn't know about this. On sio2.mimuw we also have 16x time limit for sio2jail